packages/flutter/lib/src/widgets/automatic_keep_alive.dart

```diff
@@ -28,42 +28,22 @@ class AutomaticKeepAlive extends StatefulWidget {
   /// [KeepAlive] widget appropriately.
   const AutomaticKeepAlive({
     Key? key,
-    this.child,
+    required this.child,
   }) : super(key: key);
 
   /// The widget below this widget in the tree.
   ///
   /// {@macro flutter.widgets.ProxyWidget.child}
-  final Widget? child;
+  final Widget child;
 
   @override
   State<AutomaticKeepAlive> createState() => _AutomaticKeepAliveState();
 }
 
 class _AutomaticKeepAliveState extends State<AutomaticKeepAlive> {
   Map<Listenable, VoidCallback>? _handles;
-  Widget? _child;
```

Comment:
@Hixie I think this can be clean up furthermore. is there a reason we try to cache this child?

Reply:
nvm it looks it somehow needs to maintain the scope of _addClient when applying parent data. removed it will cause test failures